php/src/class-plugin-settings.php

```diff
@@ -0,0 +1,569 @@
+	}
+
+	/**
+	 * Render the Web Vitals event menta dimension setting field.
+	 *
+	 * @return void
+	 */
+	public function render_web_vitals_field_dimension_event_meta() {
+		$value = $this->get_web_vitals_dimension_event_meta();
+		$theme_setting_value = $this->theme_settings->get( self::SETTINGS_VITALS_FIELD_DIMENSION_EVENT_META );
+
+		if ( ! empty( $theme_setting_value ) ) {
+			$value = $theme_setting_value;
+		}
+
+		?>
+		<input
+			type="text"
+			class="regular-text"
+			name="<?php echo esc_attr( $this->get_web_vitals_setting_field_name( self::SETTINGS_VITALS_FIELD_DIMENSION_EVENT_META ) ); ?>"
+			pattern="dimension[0-9]{1,}"
+			value="<?php echo esc_attr( $value ); ?>"
+			placeholder="dimension2"
+			<?php disabled( ! empty( $theme_setting_value ) ); ?>
+		/>
+		<p class="description">
+			<?php esc_html_e( 'Specify the custom dimension used for sending the event meta data. Defaults to dimension2.' ); ?>
+		</p>
+		<?php
+	}
+
+	/**
+	 * Get the Web Vitals debug dimension.
+	 *
+	 * @return string|null
+	 */
+	public function get_web_vitals_dimension_event_debug() {
+		return $this->get_web_vitals_setting_with_theme_support_override( self::SETTINGS_VITALS_FIELD_DIMENSION_EVENT_DEBUG );
+	}
+
+	/**
+	 * Render the debug event dimension setting field.
+	 *
+	 * @return void
+	 */
+	public function render_web_vitals_field_dimension_event_debug() {
+		$value = $this->get_web_vitals_dimension_event_debug();
+		$theme_setting_value = $this->theme_settings->get( self::SETTINGS_VITALS_FIELD_DIMENSION_EVENT_DEBUG );
+
+		if ( ! empty( $theme_setting_value ) ) {
+			$value = $theme_setting_value;
+		}
+
+		?>
+		<input
+			type="text"
+			class="regular-text"
+			name="<?php echo esc_attr( $this->get_web_vitals_setting_field_name( self::SETTINGS_VITALS_FIELD_DIMENSION_EVENT_DEBUG ) ); ?>"
+			pattern="dimension[0-9]{1,}"
+			value="<?php echo esc_attr( $value ); ?>"
+			placeholder="dimension3"
+			<?php disabled( ! empty( $theme_setting_value ) ); ?>
+		/>
+		<p class="description">
+			<?php esc_html_e( 'Analytics dimension for sending debug data. Defaults to dimension3.' ); ?>
+		</p>
+		<?php
+	}
+
+	/**
+	 * Get the tracking ratio.
+	 *
+	 * @return float
+	 */
+	public function get_web_vitals_tracking_ratio() {
+		$setting = $this->get_web_vitals_setting( self::SETTINGS_VITALS_FIELD_TRACKING_RATIO );
+
+		if ( has_filter( 'site_performance_tracker_chance' ) ) {
+			$setting->set(
+				apply_filters( 'site_performance_tracker_chance', self::TRACKING_DEFAULT_CHANCE )
+			);
+		}
+
+		return $setting->get();
+	}
+
+	/**
+	 * Ensure the tracking ratio can be stored as empty to
+	 * use the default value.
+	 *
+	 * @param mixed $value Value to be saved.
+	 *
+	 * @return float|null
+	 */
+	public function sanitize_web_vitals_field_tracking_ratio( $value ) {
+		$value = trim( $value );
+
+		if ( is_numeric( $value ) ) {
+			return floatval( max( min( $value, 1 ), 0 ) );
```

Comment:
PHP4 style casts should be avoided:
```suggestion
			return (float) max( min( $value, 1 ), 0 );
```